Python/Include/Scalar.hpp

```diff
@@ -0,0 +1,173 @@
+#ifndef ROVER_PYTHON_SCALAR_HPP
+#define ROVER_PYTHON_SCALAR_HPP
+#include <type_traits>
+#include <dlib/algs.h>
+#include <pybind11/pybind11.h>
+
+namespace Rover {
+
+  //! PythonScalar forward declaration.
+  template<typename T>
+  class PythonScalar;
+
+  //! Pointer type for PythonScalar convertible to both T* and
+  //! PythonScalar<T>*.
+  /*
+    \tparam T The type of the scalar.
+  */
+  template<typename T>
+  class PythonScalarPointer {
+    public:
+
+      //! Constructs the object using a raw PythonScalar<T> pointer.
+      PythonScalarPointer(PythonScalar<T>* ptr);
+
+      //! Converts the pointer to PythonScalar<T>*.
+      operator PythonScalar<T>*() const;
```

Comment:
Similarly, should these conversions be ```explicit``` or do you intend to support implicit conversions? Generally we prefer conversions to be explicit.